src/vmm/src/builder.rs

```diff
@@ -723,16 +729,26 @@ fn attach_block_devices(
 
 fn attach_net_devices(
     vmm: &mut Vmm,
-    network_ifaces: &NetBuilder,
+    net_builder: &mut NetBuilder,
+    mmds_config: Option<&MmdsConfig>,
     event_manager: &mut EventManager,
 ) -> std::result::Result<(), StartMicrovmError> {
     use self::StartMicrovmError::*;
 
-    for net_device in network_ifaces.iter() {
+    for net_device in net_builder.iter_mut() {
         event_manager
             .add_subscriber(net_device.clone())
             .map_err(RegisterEvent)?;
-
+        match mmds_config {
+            None => (),
+            Some(cfg) => {
+                cfg.ipv4_addr().map_or((), |ipv4_addr| {
+                    if let Some(mmds_ns) = net_device.lock().unwrap().mmds_ns() {
+                        mmds_ns.set_ipv4_addr(ipv4_addr);
+                    };
+                });
+            }
+        };
```

Comment:
I think this would be better placed in the `resources` part instead of here since:

- it's not really part of _attaching_ a `Net` device to a `Vmm`.
- locality is better if you set the data as soon as you get it instead of deferring to a later step

What do you think?